ingot-macros/src/choice.rs

```diff
@@ -289,6 +295,30 @@ pub fn attr_impl(attr: TokenStream, item: syn::ItemEnum) -> TokenStream {
             }
         }
 
+        impl<V: ::ingot::types::ByteSlice> ::ingot::types::NextLayerChoice for #validated_ident<V>
+        {
+            type Hint = #on;
+
+            #[inline]
+            fn next_layer_choice(&self, _hint: Option<Self::Hint>) -> ::core::option::Option<Self::Denom> {
```

Comment:
Nit -- macro hygiene:
```suggestion
            fn next_layer_choice(&self, _hint: ::core::option::Option<Self::Hint>) -> ::core::option::Option<Self::Denom> {
```

This comes up a few times here and in `ingot-macros/src/packet/mod.rs`.

Reply:
Good catch, fixed!